src/Common/Core/Impl/OS/ProcessExtensions.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Common.Core.Tasks;
+
+namespace Microsoft.Common.Core.OS {
+    public static class ProcessExtensions {
+        public static Task WaitForExitAsync(this IProcess process, int milliseconds, CancellationToken cancellationToken = default(CancellationToken)) {
+            var tcs = new TaskCompletionSource<int>();
+            process.Exited += (o, e) => tcs.TrySetResult(0);
```

Comment:
> Is there a way to know if this returned due to time out or because the process exited?

Async methods throw `OperationCanceledException` when they fail by timeout, so if task is completed, it has actually exited.